### PR TITLE
Use buttons for the bookmark command

### LINKED ISSUE #932: Replace reactions in the `bookmark` command with buttons.
## Description
Right now, bookmarking messages as someone who is not the command invoker requires adding a reaction. I believe that the system can be made much more better using buttons as it greatly improves the UI and makes it a bit more simpler.

## Reasoning
While using interactions, we now have an option to make any error only visible to the interaction author by making the error message ephemeral. This would mean that the error message will not be publicly visible and a confirmation message that is only visible to the interaction author can be sent as well.

## Proposed Implementation
image

## Additional Details
We can even make use of the link buttons and attach them to the bookmarked message. When clicked, they would redirect the user to the message. Adding a button for it would make the embed solely limited to the message content and a `View Message` link button would improve the UI as well as make navigation to the message a bit more easier.

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

### PR DESCRIPTION
**Pull request by Diabolical5777**
## Relevant Issues

Closes #932

## Description

I replaced the reactions used currently with buttons

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/utilities/bookmark.py [hunk 1]
```diff
@@ -25,6 +23,53 @@
 )
 
 
+class LinkTargetMessage(discord.ui.View):
+    """The button that relays the user to the bookmarked message."""
+
+    def __init__(
+        self,
+        target_message: discord.Message
```
Comment (Dorukyum) [inline bot/exts/utilities/bookmark.py:31]: Let's add a trailing comma if we choose to use this paranthesis syntax
```suggestion
        target_message: discord.Message,
```

File: bot/exts/utilities/bookmark.py [hunk 2]
```diff
@@ -25,6 +23,53 @@
...
+
+        self.bookmark_function = bookmark_function
+        self.clicked: list[int] = [author.id]
+        self.channel = channel
+        self.target_message = target_message
+
+        super().__init__(label="Receive Bookmark", style=discord.ButtonStyle.green)
+
+    async def callback(self, interaction: discord.Interaction) -> None:
+        """The callback to the function."""
```
Comment (Dorukyum) [inline bot/exts/utilities/bookmark.py:58]: Maybe to be more clear?
```suggestion
        """The button callback."""
```

File: bot/exts/utilities/bookmark.py (hunk 1 above)
Comment (Diabolical5777) [reply bot/exts/utilities/bookmark.py:31]: I considered that but to be consistent with how they make the init currently, I left it out. And I am not sure what you mean by paranthesis syntax

Comment (Dorukyum) [reply bot/exts/utilities/bookmark.py]: I mean we don't really need to seperate this into multiple lines, line below is 56 chars:
```python
    def __init__(self, target_message: discord.Message):
```

**Comment by Robin5605** [comment]
It looks like we have a full on `discord.ui.View` subclass for a simple link button - but for the actual button that sends the bookmark in the DM, we're just doing:
```py
view = discord.ui.View(timeout=TIMEOUT)
view.add_item(
    SendBookmark(self.action_bookmark, ctx.author, ctx.channel, target_message)
)
```
I think this should be flipped around. There's no need for a full fledged subclass for a simple link button, but maybe we can use a view subclass for the actual bookmark. I doubt we'll also need a button subclass since we're not doing anything dynamic

**Comment by Diabolical5777** [comment]
> It looks like we have a full on `discord.ui.View` subclass for a simple link button - but for the actual button that sends the bookmark in the DM, we're just doing:
> 
> ```python
> view = discord.ui.View(timeout=TIMEOUT)
> view.add_item(
>     SendBookmark(self.action_bookmark, ctx.author, ctx.channel, target_message)
> )
> ```
> 
> I think this should be flipped around. There's no need for a full fledged subclass for a simple link button, but maybe we can use a view subclass for the actual bookmark. I doubt we'll also need a button subclass since we're not doing anything dynamic

I think you have it flipped. I used the button subclass for the button that is sent in the channel where the bookmark is invoked in so others can press the button. For the button in DM's that redirects to the bookmarked message, I used a view directly.

**Comment by Robin5605** [comment]
You're right that I had it flipped. The only other thing I might recommend we change is maybe subclassing `discord.ui.View` instead of a `discord.ui.Button`? Usually that's what we do for non-dynamic components. And I think all the parameters we pass into `SendBookmark` can be passed into the view subclass, and we shouldn't lose any functionality

**Pull request closed without merging**